Updated README.md (#646)
holochain/holochain: README.md

This nix-shell flavor installs wrapper binaries for `holochain` and `hc` that will automatically compile and run the binaries.  This is very useful if you are tracking changes in the holochain repo because when you check out a new rev, running holochain will compile automatically to the version at that rev.

> ### Building with cargo:

Reviewer: ```suggestion
### Building with a pre-installed cargo
```

> Another way to install the `holochain` and `dna-util` binaries (if the previous didn't work) is by using rust

Reviewer: ```suggestion
Another way to install the `holochain` and `hc` binaries (if the previous didn't work) is by using rust
```

> [Install Rust](https://www.cloudbooklet.com/install-rust-on-ubuntu-18-04-lts/) 

Reviewer 2: ```suggestion
[Install Rust](https://www.rust-lang.org/tools/install) 
```
Maybe we can link to the official install instructions?

Author: By default I was inclined to do so, but this manual is more complete and detailed than the official.

Reviewer 2: My concern is that it will go out of date and is aimed specifically at ubuntu-18-04

Author: ok

Install holochain binaries:
```
cargo install --path crates/holochain
cargo install --path crates/dna_util
```

> be sure to add `holochain/.cargo/bin` to your PATH to be able to run the installed binaries

Reviewer 3: On my computer, `cargo install` puts things in `~/.cargo/bin` and puts the installed copy of holochain in `~/.cargo/bin/holochain`.

```suggestion
be sure to add `~/.cargo/bin` to your PATH to be able to run the installed binaries
```

Reviewer: I think this stems from the nix-shell changing the ` CARGO_TARGET_DIR` to the root of the holochain directory. Since this chapter is about using the pre-installed cargo toolchain I agree with @timotree3's suggestion here
